gadgets/src/integers/alloc.rs

```diff
@@ -35,33 +35,44 @@ macro_rules! alloc_gadget_fn_impl {
             mut cs: CS,
             value_gen: Fn,
         ) -> Result<Self, SynthesisError> {
-            let value = value_gen().map(|val| *val.borrow());
-            let values = match value {
-                Ok(mut val) => {
-                    let mut v = Vec::with_capacity(<$gadget as Integer>::SIZE);
+            if cs.is_in_setup_mode() {
```

Comment:
Now, when allocating Vec<Boolean>, it will not retrieve the value when it is in the setup mode. This is crucial for the Div gadget to be runnable in the setup mode.